src/node/serialization.cc

```diff
@@ -414,29 +415,39 @@ std::string SaveJSON(const ObjectRef& n) {
 }
 
 ObjectRef LoadJSON(std::string json_str) {
-  std::istringstream is(json_str);
-  dmlc::JSONReader reader(&is);
   JSONGraph jgraph;
-  // load in json graph.
-  jgraph.Load(&reader);
-  std::vector<ObjectPtr<Object> > nodes;
+  std::vector<ObjectPtr<Object>> nodes;
   std::vector<runtime::NDArray> tensors;
-  // load in tensors
-  for (const std::string& blob : jgraph.b64ndarrays) {
-    dmlc::MemoryStringStream mstrm(const_cast<std::string*>(&blob));
-    support::Base64InStream b64strm(&mstrm);
-    b64strm.InitPosition();
-    runtime::NDArray temp;
-    CHECK(temp.Load(&b64strm));
-    tensors.emplace_back(temp);
+  {
+    // load in json graph.
+    std::istringstream is(json_str);
+    dmlc::JSONReader reader(&is);
+    jgraph.Load(&reader);
+    // load in tensors
+    for (const std::string& blob : jgraph.b64ndarrays) {
+      dmlc::MemoryStringStream mstrm(const_cast<std::string*>(&blob));
+      support::Base64InStream b64strm(&mstrm);
+      b64strm.InitPosition();
+      runtime::NDArray temp;
+      CHECK(temp.Load(&b64strm));
+      tensors.emplace_back(temp);
+    }
   }
   ReflectionVTable* reflection = ReflectionVTable::Global();
 
   // node 0 is always null
   nodes.reserve(jgraph.nodes.size());
 
   for (const JSONNode& jnode : jgraph.nodes) {
-    if (jnode.type_key.length() != 0) {
+    if (jnode.type_key == ArrayNode::_type_key) {
+      CHECK(jnode.repr_bytes.empty());
+      int64_t size = jnode.data.size();
+      ObjectPtr<ArrayNode> node = ArrayNode::make(size);
+      for (int64_t& i = node->size_ = 0; i < size; ++i) {
+        node->EmplaceInit(i, nullptr);
```

Comment:
Can we simply do ArrayNode::Create(size, ObjectRef(nullptr)) here?

Reply:
Makes sense. Will do!